pkg/controller/nodemaintenance/nodemaintenance_controller.go

```diff
@@ -254,6 +256,37 @@ func (r *ReconcileNodeMaintenance) Reconcile(request reconcile.Request) (reconci
 	return reconcile.Result{}, nil
 }
 
+
+func setOwnerRefToNode(instance *nodemaintenanceapi.NodeMaintenance, node *corev1.Node) (bool) {
+
+	objectRefIsSet := false
+	for _, ref := range instance.ObjectMeta.GetOwnerReferences() {
+		if ref.APIVersion == node.TypeMeta.APIVersion && ref.Kind == node.TypeMeta.Kind && ref.Name == node.ObjectMeta.GetName() && ref.UID == node.ObjectMeta.GetUID() {
+			objectRefIsSet = true
+			break
+		}
+	}
+
+	if !objectRefIsSet {
+		log.Info("setting owner ref to node")
+
+		blockOwnerDeletion := false
+		controllerDeletion := false
+		nodeMeta := node.TypeMeta
+		ref := metav1.OwnerReference{
+				APIVersion:         nodeMeta.APIVersion,
+				Kind:               nodeMeta.Kind,
+				Name:               node.ObjectMeta.GetName(),
+				UID:                node.ObjectMeta.GetUID(),
+				BlockOwnerDeletion: &blockOwnerDeletion, // false - deletion of node doesn't have to wait for nmo CR deletion
+				Controller:         &controllerDeletion, // false - can delete nmo without deleting the node
+			}
+
+		instance.ObjectMeta.SetOwnerReferences( append(instance.ObjectMeta.GetOwnerReferences(), ref) )
+	}
+	return !objectRefIsSet
```

Comment:
The return value isn't evaluated in the actual code, so it can be removed.
You should not add a return value just for the test IMHO.

Reply:
done